pages/wallet.js

```diff
@@ -279,7 +280,13 @@ function InvoiceScanner ({ fieldName }) {
           return (
             <QrScanner
               onDecode={(result) => {
-                helpers.setValue(result.replace(/^lightning:/, '').toLowerCase())
+                if (result.split('lightning=')[1]) {
+                  helpers.setValue(result.split('lightning=')[1].split(/[&?]/)[0].toLowerCase())
+                } else if (!result.split('lightning=')[1] && (decode(result.replace(/^lightning:/, '')) || decode(result))) {
```

Comment:
Excellent! 

We don't need to test for `!result.split('lightning=')[1]` given this is an `else` branch and `if` will be taken when `result.split('lightning=')[1]` is truthy.

Reply:
Ok should be good! thanks for your patience with me on this one! getting a hang of parsing bolt11 made my head spin :dizzy_face: